qgreenland/models/config/layer_group.py

```diff
@@ -1,19 +1,104 @@
+import re
+from collections import UserString
 from typing import Optional, Union
 
 from qgreenland.models.base_model import QgrBaseModel
 
+layer_identifier_regex = re.compile(r"[a-z0-9_]*")
+layer_group_identifier_regex = re.compile(r"[A-Z0-9][a-zA-Z0-9() ,._-]*")
+
+
+class LayerIdentifier(UserString):
+    """A string uniquely identifying a layer.
+
+    Based on docs: https://docs.pydantic.dev/1.10/usage/types/#classes-with-__get_validators__
+
+    TODO: Update other models to replace string layer identifiers with this class?
+    """
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    # TODO: __modify_schema__?
```

Comment:
What does this do? More context or remove maybe?

Reply:
e27f2941